NOTE-act-rules-common-aspects.bs

```diff
@@ -68,4 +68,21 @@ Language, either written or spoken, contained in nodes of the DOM [[DOM]] or acc
 
 Rules can only operate on the Language aspect if the language of a page can be determined. The means by which the language is assessed, whether by a person or a machine, is not of importance as long as the assessment meets the criteria defined in [[wcag2-tech-req#humantestable]] [[WCAG]].
 
-Some rules can only operate on a Language aspect if the language is sufficiently understood, while other rules only require identifying the language. For example, a rule checking that an accessible name is descriptive can only function if the language is understood. A rule checking the correctness of a `lang` attribute requires knowing what language is used, but not the meaning of the words.
+Some rules can only operate on a Language aspect if the language is sufficiently understood by the tester, while other rules only require identifying the language. For example, a rule checking that an accessible name is descriptive can only function if the language is understood. A rule checking the correctness of a `lang` attribute requires knowing what language is used, but not the meaning of the words.
+
+Source Code {#input-aspects-code}
+----------------------------------
+
+The text content of a file from which a web browser or other user agent creates a page. For example, a browser may build up a web page from an HTML file, CSS file, SVG file and JavaScript file. The text of each of these four files is its source code. An ACT Rule could for example test for potential parser errors.
+
+Source code is distinct from an HTTP response, which includes HTTP headers. It is the text content of the file before parsing, which often results in an object model or syntax tree, or serialized versions of those. Notably in HTML, the outerHTML property of the root node can vary significantly from its source code. For the purpose of ACT rules, source files used in preprocessing such as PHP for creating HTML, or SASS for creating CSS are not considered, as those are not processed by the user agent.
```

Comment:
Wondering if the last sentence (preprocessing vs user-agent) could be phrased (instead or in addition to the current one) in term of server/client processing?
Something like precising "preprocessing" as "preprocessing server side" or a similar sentence. One could argue that client side scripting that produce HTML is also preprocessing, but I guess we do want to consider it as "source code" 🤔

Reply:
So then how about this:

```suggestion
Source code is distinct from an HTTP response, which includes HTTP headers. It is the text content of the file before parsing, which often results in an object model or syntax tree, or serialized versions of those. Notably in HTML, the outerHTML property of the root node can vary significantly from its source code. For the purpose of ACT rules, source files processed on the server such as PHP for creating HTML, or SASS for creating CSS are not considered, as those are not processed by the user agent.```

Reply:
I like it 👍